### ISSUE TITLE
[java] DataflowAnomalyAnalysis has false positive for object initialised outside loop.

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2131]
**Affects PMD Version:** 
6.19.0

**Rule:**
DataflowAnomalyAnalysis.

**Description:**
False positive for variable initialised outside loop.
The error occurs in all three methods.

**Code Sample demonstrating the issue:**

```java
 package pmdtests;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class TestDu {

  private List<String> list = new ArrayList<>();

  public void run() {
    String str = Thread.currentThread().getName() + " Element : %d";
    for (int i = 0; i < 10_000; i++) {
      list.add(String.format(str, i));
    }
  }

  public void runAgain() {
    String str = Thread.currentThread().getName() + " Element : %d";
    for (int i = 0; i < 10_000; i++)
      list.add(String.format(str, i));
  }

  public void runOnceMore() {
    String str = Thread.currentThread().getName() + " Element : %d";
  list =  IntStream.range(0, 10_000)
        .mapToObj(i -> String.format(str, i))
        .collect(Collectors.toList());
  }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by adangel** [comment]
The rule [DataflowAnomalyAnalysis](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#dataflowanomalyanalysis) has been deprecated in favour of [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment), which is available with PMD 6.26.0.

Therefore this issue wont be fixed.
